File: starter_kits/Java/hlt/Game.java
package hlt;

import java.util.ArrayList;
import java.util.Collection;

public class Game {
    public int turnNumber;
    public final PlayerId myId;
    public final ArrayList<Player> players = new ArrayList<>();
    public final Player me;
    public final GameMap gameMap;

    public Game() {
        Constants.populateConstants(Input.readLine());

        final Input input = Input.readInput();
        final int numPlayers = input.getInt();
        myId = new PlayerId(input.getInt());

        Log.open(myId.id);

        for (int i = 0; i < numPlayers; ++i) {
            players.add(Player._generate());
        }
        me = players.get(myId.id);
        gameMap = GameMap._generate();
    }

    public void ready(final String name) {
        System.out.println(name);
    }

    public void updateFrame() {
        turnNumber = Input.readInput().getInt();
        Log.log("=============== TURN " + turnNumber + " ================");

        for (int i = 0; i < players.size(); ++i) {
            final Input input = Input.readInput();

            final PlayerId currentPlayerId = new PlayerId(input.getInt());
            final int numShips = input.getInt();
            final int numDropoffs = input.getInt();
            final int halite = input.getInt();

            players.get(currentPlayerId.id)._update(numShips, numDropoffs, halite);
        }

        gameMap._update();

        for (final Player player : players) {
            for (final Ship ship : player.ships.values()) {
                gameMap.at(ship).markUnsafe(ship);
            }

            gameMap.at(player.shipyard).structure = player.shipyard;

            for (final Dropoff dropoff : player.dropoffs.values()) {
                gameMap.at(dropoff).structure = dropoff;
            }
        }
    }

    public void endTurn(final Collection<Command> commands) {
        for (final Command command : commands) {
            System.out.print(command.command);
            System.out.print(' ');
        }
        System.out.println();
    }
}

File: starter_kits/Kotlin/hlt/Game.kt
package hlt

import java.util.*

class Game {
    var turnNumber: Int = 0
    val myId: PlayerId
    val players: ArrayList<Player> = ArrayList()
    val me: Player
    val gameMap: GameMap

    init {
        Constants.populateConstants(Input.readLine())

        val input = Input.readInput()
        val numPlayers = input.nextInt
        myId = PlayerId(input.nextInt)

        Log.open(myId.id)

        for (i in 0 until numPlayers) {
            players.add(Player._generate())
        }
        me = players.get(myId.id)
        gameMap = GameMap._generate()
    }

    fun ready(name: String) {
        System.out.println(name)
    }

    fun updateFrame() {
        turnNumber = Input.readInput().nextInt
        Log.log("=============== TURN $turnNumber ================")

        for (i in 0 until players.size) {
            val input = Input.readInput()

            val currentPlayerId = PlayerId(input.nextInt)
            val numShips = input.nextInt
            val numDropoffs = input.nextInt
            val halite = input.nextInt

            players.get(currentPlayerId.id)._update(numShips, numDropoffs, halite)
        }

        gameMap._update()

        for (player in players) {
            for (ship in player.ships.values) {
                gameMap.at(ship)?.markUnsafe(ship)
            }

            gameMap.at(player.shipyard)?.structure = player.shipyard

            for (dropOff in player.dropoffs.values) {
                gameMap.at(dropOff)?.structure = dropOff
            }
        }
    }

    fun endTurn(commands: Collection<Command>) {
        for (command in commands) {
            System.out.print(command.command)
            System.out.print(' ')
        }
        System.out.println()
    }
}
